modules/nlu/src/backend/api.ts

```diff
@@ -15,28 +20,58 @@ export const PredictSchema = Joi.object().keys({
   text: Joi.string().required()
 })
 
-export default async (bp: typeof sdk, state: NLUState) => {
-  const router = bp.http.createRouterForBot('nlu')
+const makeErrorMapper = (bp: typeof sdk) => (err: { botId: string; lang: string; error: Error }, res: Response) => {
+  const { error, botId, lang } = err
+
+  if (error instanceof BotNotMountedError) {
+    return res.status(404).send(`Bot ${botId} doesn't exist`)
+  }
+
+  if (error instanceof BotDoesntSpeakLanguageError) {
+    return res.status(422).send(`Language ${lang} is either not supported by bot or by language server`)
+  }
+
+  const msg = 'An unexpected error occured.'
+  bp.logger
+    .forBot(botId)
+    .attachError(error)
+    .error(msg)
+  return res.status(500).send(msg)
+}
+
+const mapTrainSession = (ts: TrainingSession): sdk.NLU.TrainingSession => {
+  const { botId, language, progress, status } = ts
+  const key = `training:${botId}:${language}`
+  return { key, language, status, progress }
+}
+
+export const getWebsocket = (bp: typeof sdk) => {
+  return async (ts: TrainingSession) => {
+    const { botId } = ts
+    const trainSession = mapTrainSession(ts)
+    const ev: NLUProgressEvent = { type: 'nlu', botId, trainSession }
+    bp.realtime.sendPayload(bp.RealTimePayload.forAdmins('statusbar.event', ev))
+  }
+}
+
+export const registerRouter = async (bp: typeof sdk, app: NLUApplication) => {
+  const router = bp.http.createRouterForBot(ROUTER_ID)
+
+  const mapError = makeErrorMapper(bp)
   const needsWriteMW = bp.http.needPermission('write', 'bot.training')
 
   router.get('/health', async (req, res) => {
-    // When the health is bad, we'll refresh the status in case it changed (eg: user added languages)
-    const health = state.engine.getHealth()
+    // When the health is bad, we'll refresh the status in case it has changed (eg: user added languages)
+    const health = app.getHealth()
```

Comment:
By the way, I think we should websocket the heck out of this as well and notify the ui if health changes